src/request.ml

```diff
@@ -895,6 +895,14 @@ value treat_request conf base = do {
   [ (Some s, _, _) -> print_moved conf base s
   | (_, Some "no_index", _) -> print_no_index conf base
   | (_, _, Some "IM") -> Image.print conf base
+  | (_, _, Some "DOC") -> 
+    match p_getenv conf.env "s" with
+    [ Some f ->
+      if String.sub f (String.length f - 4) 4 = ".txt" then
```

Comment:
Better to use `Filename.check_suffix` and then `Filename.chop_suffix`
https://caml.inria.fr/pub/docs/manual-ocaml/libref/Filename.html